plugins/woocommerce/includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -392,4 +487,115 @@ protected function prepare_column_headers() {
 			$this->get_sortable_columns(),
 		);
 	}
+
+	/**
+	 * Convert an array or object into a multidimensional array.
+	 *
+	 * @param array|object $input Array or object to be processed.
+	 *
+	 * @return array
+	 */
+	public function to_multidimensional_array( $input ) {
+		$out   = array();
+		$input = is_object( $input ) ? (array) $input : $input;
+
+		if ( is_array( $input ) ) {
+
+			// Run function on each item in the array.
+			$out = array_map(
+				function( $value ) {
+
+					if ( is_array( $value ) || is_object( $value ) ) {
+						return $this->to_multidimensional_array( $value );
+					}
+
+					return $value;
+				},
+				$input
+			);
+		}
+
+		return $out;
+	}
+
+	/**
+	 * Convert an array to a string with tabbed indentaion.
+	 *
+	 * @param array|object $input   An object or array.
+	 * @param array        $exclude An array of strings of params/keys to exclude.
+	 * @param int          $depth   The current depth of the walker.
+	 *
+	 * @return string
+	 */
+	public function printable_array( $input, $exclude = array(), $depth = 0 ) {
```

Comment:
I wonder if, rather than add here, this could live in a new utility class (perhaps `Automattic\WooCommerce\Internal\Utilities\Arrays`)?

That would make it easier to use from elsewhere, and would avoid us expanding the public surface area of this class more than needed.